.github/scripts/set_test_package_matrix.sh

```diff
@@ -5,7 +5,17 @@
 set -euo pipefail
 export RUNNER_COUNT=$1
 
+if ((RUNNER_COUNT == 1 ))
+then
+  EFFECTIVE_RUNNER_COUNT=$RUNNER_COUNT
+elif ((RUNNER_COUNT > 1 ))
+then
+  EFFECTIVE_RUNNER_COUNT=$((RUNNER_COUNT-1))
+else
+  echo ERROR: RUNNER_COUNT must be greater than zero.
+  exit 1 # terminate and indicate error
+fi
 # set matrix var to list of unique packages containing tests
-matrix="$(go list -json="ImportPath,TestGoFiles" ./... | jq --compact-output '. | select(.TestGoFiles != null) | .ImportPath' | shuf | jq --slurp --compact-output '.' | jq --argjson runnercount $RUNNER_COUNT  -cM '[_nwise(length / $runnercount | floor)]'))"
+matrix="$(go list -json="ImportPath,TestGoFiles" ./... | jq --compact-output '. | select(.TestGoFiles != null) | .ImportPath' | shuf | jq --slurp --compact-output '.' | jq --argjson runnercount $EFFECTIVE_RUNNER_COUNT  -cM '[_nwise(length / $runnercount | floor)]'))"
 
-echo "matrix=${matrix}" >> "${GITHUB_OUTPUT}"
+echo "matrix=${matrix}" >> "${GITHUB_OUTPUT}"
```

Comment:
```suggestion
export BUCKET_COUNT=$1

if ((BUCKET_COUNT == 0)); then
  echo ERROR: BUCKET_COUNT must be greater than zero.
  exit 1 # terminate and indicate error
fi

test_packages="$(go list -json="ImportPath,TestGoFiles" ./... | jq --compact-output '. | select(.TestGoFiles != null) | .ImportPath' | shuf | jq --slurp --compact-output '.')"

# Ensure buckets match desired count (_nwise will add one if BUCKET_COUNT doesn't divide tests evenly)
if ((BUCKET_COUNT % $(echo $test_packages | jq 'length') )); then
  BUCKET_COUNT=$((BUCKET_COUNT-1))
fi

matrix=$(echo $test_packages | jq --argjson runnercount $BUCKET_COUNT -cM '[_nwise(length / $runnercount | floor)]')
```

I think `_nwise` is actually variable in the length of its output based on whether there's a "remainder":
```shell-session
❯ jq -nc '[1,2,3,4,5,6,7,8,9,10] | _nwise(length / 3 | floor)'
[1,2,3]
[4,5,6]
[7,8,9]
[10]
~/workspace                                                                                                                                                                22:22:05
❯ jq -nc '[1,2,3,4,5,6,7,8,9] | _nwise(length / 3 | floor)'
[1,2,3]
[4,5,6]
[7,8,9]
~/workspace                                                                                                                                                                22:22:12
❯ jq -nc '[1,2,3,4,5,6,7,8] | _nwise(length / 3 | floor)'
[1,2]
[3,4]
[5,6]
[7,8]
~/workspace                                                                                                                                                                22:22:19
❯ jq -nc '[1,2,3,4,5,6,7] | _nwise(length / 3 | floor)'
[1,2]
[3,4]
[5,6]
[7]
~/workspace                                                                                                                                                                22:22:37
❯ jq -nc '[1,2,3,4,5,6] | _nwise(length / 3 | floor)'
[1,2]
[3,4]
[5,6]
```

Looks like a modulo test will make sure it works either way.

Reply:
Oh, actually, it looks like if we just swap `floor` for `ceil`, that solves it:

```shell-session
❯ jq -nc '[1,2,3,4,5,6,7,8,9,10] | _nwise(length / 3 | ceil)'
[1,2,3,4]
[5,6,7,8]
[9,10]
❯ jq -nc '[1,2,3,4,5,6,7,8,9] | _nwise(length / 3 | ceil)'
[1,2,3]
[4,5,6]
[7,8,9]
❯ jq -nc '[1,2,3,4,5,6,7,8] | _nwise(length / 3 | ceil)'
[1,2,3]
[4,5,6]
[7,8]
❯ jq -nc '[1,2,3,4,5,6,7] | _nwise(length / 3 | ceil)'
[1,2,3]
[4,5,6]
[7]
❯ jq -nc '[1,2,3,4,5,6] | _nwise(length / 3 | ceil)'
[1,2]
[3,4]
[5,6]
```